### PR TITLE
Straighten backends

### PR DESCRIPTION
**Pull request by michalhabera**
Rearrange files from `ffc/backends` into `ffc/codegeneration`.

### CODE REVIEW

File: ffc/codegeneration/__init__.py [hunk 1]
```diff
@@ -1,13 +1,16 @@
 # -*- coding: utf-8 -*-
-"""Utility functions for UFC"""
-
-__author__ = "FEniCS Project"
-__version__ = "2018.2.0.dev0"
-__license__ = "This code is released into the public domain"
+# Copyright (C) 2011-2018 Michal Habera and Martin Sandve Alnæs
```
Comment (garth-wells) [inline ffc/codegeneration/__init__.py]: Need to be careful changing the license here.

Comment (michalhabera) [reply ffc/codegeneration/__init__.py]: Just copied from some other file and added my name. What should I do? I do not insist on my name to be there.

Comment (garth-wells) [reply ffc/codegeneration/__init__.py]: Adding name is fine if you've made changes. Issue is the license change which can't be done without permission of all copyright holders.

Comment (michalhabera) [reply ffc/codegeneration/__init__.py]: @garth-wells Fixed

**Pull request merged**